Rebaseline to 2023 Data

JamesCribb-TfN opened the pull request:
Changes:
1. Updated relevant classes for 2023 re-baseline.
2. Further changes to load_data.py by reading from .env file and now supports linux/windows path
3. General tables and Scenario tables read in differently
4. Remove unused code

Actions:
1. test run_fresh=True
2. run full CafCarb with all scenarios and compare with previous runs
3. utility.py TODO update method of reading table ranges
5. Unit Tests

JamesCribb-TfN committed:
Rebaseline to 2023 Data

1. Updated relevant classes for 2023 re-baseline.
2. Further changes to load_data.py by reading from .env file and now supports linux/windows path
3. General tables and Scenario tables read in differently

Actions:
1. test run_fresh=True
2. run full CafCarb with all scenarios
3. utility.py TODO update method of reading table ranges

JamesCribb-TfN committed:
remove unused code in utility.py

JamesCribb-TfN committed:
isort and black to format files

JamesCribb-TfN commented:
working through actions :)

BenTaylor-TfN commented on .github/workflows/isort.yml:
```diff
@@ -2,15 +2,21 @@ name: Isort Formatting
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+      - name: Set up Node.js 20.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20.x'
```
What are these changes?

JamesCribb-TfN replied:
Sets up the Node.js environment with version 20.x for the GitHub Actions. They're going to depreciate the previous one. Just got rid of a couple of depreciation errors (still some more :) )

BenTaylor-TfN replied:
```suggestion
```

BenTaylor-TfN commented on requirements_dev.txt:
```diff
@@ -12,3 +12,10 @@ pytest-xdist>=3.3.1
 
 # edit_install
 versioningit>=2.2.0, < 3.0.0
+
+pandas~=2.2.1
+python-dotenv~=1.0.1
+numpy~=1.26.4
+matplotlib~=3.8.3
+scikit-learn~=1.4.1.post1
+pydantic~=2.6.3
```
Are these needed for development rather than running the model?

Requirements in here should be extra needed on top of those in `requirements.txt`, and only needed for those developing the tools rather than running.

JamesCribb-TfN replied:
No they are needed for the model. These are for the requirments.txt - will change

JamesCribb-TfN replied:
use >= rather than ~= or ==- 
base version is the one that I've been using

BenTaylor-TfN commented on .gitignore:
```diff
@@ -20,6 +20,7 @@ __pycache__/
 *.xlsx
 *.pbz2
 *.docx
+.env
```
Are .env files being used here?

Environments should be used via anaconda / miniforge.

JamesCribb-TfN replied:
I've always used .env files to store hardcoded sensitive info - API keys, DB info, Filepaths etc. I think its a bit less cumbersome than config files for portability. Will leave this call up to you. Can change if needed.

JamesCribb-TfN replied:
move over to yaml

BenTaylor-TfN commented on requirements.txt:
```diff
@@ -2,4 +2,5 @@ matplotlib==3.7.3
-tables==3.8.0
\ No newline at end of file
+tables==3.8.0
+python-dotenv~=1.0.1
```
What is this requirement?

JamesCribb-TfN replied:
not sure what the tables was for. The dotenv is used to use the .env files to load the hardcoded filepaths

BenTaylor-TfN commented on template.env:
Is this essentially a config file?
I would avoid the `.env` name as it's already used for python virtual environments and could cause confusion.

I would also recommend taking a look at [`BaseConfig`](https://caftoolkit.readthedocs.io/en/stable/api/config_base.BaseConfig.html#config_base.BaseConfig) as it looks like it does something similar already

JamesCribb-TfN replied:
Yes its like a config file to store file paths. I think its a bit less cumbersome and easily ported as .env files can just be shared easily and never had problems with them (unless they're accidently pushed to repo). Mentioned in other comments.

BenTaylor-TfN reviewed (changes requested):
Quick review of package structure

AGoodchild-TfN committed:
2023 fleet demand factors

Splitting of fleet licensers by fractions to preserve NAEI vehicle classification proportions

AGoodchild-TfN committed:
2023 rebaseline changes

Caitlin-Stokes committed:
Merge branch 'fleet-emissions-tool' of E:\GitHub\caf.carbon with conflicts.
